src/shared/lib/GenericAssayUtils/TreatmentUtils.ts

```diff
@@ -0,0 +1,114 @@
+            }
+        )
+        treatments = _.map(treatmentsMeta, (meta) => {
+            return {
+                treatmentId: meta.stableId,
+                name: "NAME" in meta.genericEntityMetaProperties ? meta.genericEntityMetaProperties["NAME"] : NOT_APPLICABLE_VALUE,
+                description: "DESCRIPTION" in meta.genericEntityMetaProperties ? meta.genericEntityMetaProperties["DESCRIPTION"] : NOT_APPLICABLE_VALUE,
+                refLink: "URL" in meta.genericEntityMetaProperties ? meta.genericEntityMetaProperties["URL"] : NOT_APPLICABLE_VALUE
+            }
+        })
+    }
+    return treatments;
+}
+
+export async function fetchTreatmentByTreatmentIds(treatmentIds: string[]) {
+    let treatments: Treatment[] = [];
+    if (treatmentIds.length > 0) {
+        const treatmentsMeta = await client.fetchGenericAssayMetaDataUsingPOST(
+            {
+                genericAssayMetaFilter: {
+                    genericAssayStableIds: treatmentIds
+                    // the Swagger-generated type expected by the client method below
+                    // incorrectly requires both molecularProfileIds and genericAssayStableIds;
+                    // use 'as' to tell TypeScript that this object really does fit.
+                } as GenericAssayMetaFilter
+            }
+        )
+        treatments = _.map(treatmentsMeta, (meta) => {
+            return {
+                treatmentId: meta.stableId,
+                name: "NAME" in meta.genericEntityMetaProperties ? meta.genericEntityMetaProperties["NAME"] : NOT_APPLICABLE_VALUE,
+                description: "DESCRIPTION" in meta.genericEntityMetaProperties ? meta.genericEntityMetaProperties["DESCRIPTION"] : NOT_APPLICABLE_VALUE,
+                refLink: "URL" in meta.genericEntityMetaProperties ? meta.genericEntityMetaProperties["URL"] : NOT_APPLICABLE_VALUE,
+            }
+        })
+    }
+    return treatments;
+}
+
+export async function fetchTreatmentData(treatmentIdsByProfile: {[molecularProfileId: string]: string[]}, sampleFilterByProfile: {[molecularProfileId: string]: IDataQueryFilter}) {
+    const params = Object.keys(treatmentIdsByProfile)
+        .map(profileId => ({
+            molecularProfileId: profileId,
+            genericAssayDataFilter: {
+                genericAssayStableIds: treatmentIdsByProfile[profileId],
+                ...sampleFilterByProfile[profileId]
+                // the Swagger-generated type expected by the client method below
+                // incorrectly requires both samples and a sample list;
+                // use 'as' to tell TypeScript that this object really does fit.
+            } as GenericAssayDataFilter
+        }
+    ));
+    const dataPromises = params.map(param => client.fetchGenericAssayDataInMolecularProfileUsingPOST(param));
+    const results = await Promise.all(dataPromises);
+    const treatmentDataResult: TreatmentMolecularData[][] = _.map(results, (result: GenericAssayData[]) => {
```

Comment:
I do not like the hard-coding of the entity type or profile type in the fronted code. In my view there should only be a `GenericAssayMolecularData` type. As a result of this I do not like the existence of a util file that is dedicated to treatments. I think there should be a `GenericAssayUtils.ts` file that handles requests of all data of type _Generic Assay_. I say this because the current implementation forced us to rework the frontend code at more places than needed and resulting in the need for carbon-copied util files that are not different in functionality but are needed because of the design choice.

Can we discuss moving towards a frontend that knows about generic assay datatype, but as little as possible of its subtypes?

Reply:
Hi Pim, I understand what you are saying.  I had actually advocated the other way around.  Lets have a meeting to discuss as I think this is an important question.